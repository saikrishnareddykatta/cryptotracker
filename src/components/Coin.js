import React from "react";
import "./Coin.css";

const Coin = (props) => {
  const { id, imageSrc, name, symbol, price, marketCap, volume, priceChange } =
    props;
  return (
    <div className="cryptoCoin">
      <img src={imageSrc} className="coinLogo" alt={`${name}-${id} symbol`} />
      <div className="coinNameWrap">
        <h1 className="coinName">{name}</h1>
        <p className="coinSymbol">{symbol.toUpperCase()}</p>
      </div>
      <p className="coinPrice">${price.toLocaleString()}</p>
      <p className="coinMarketCap">Market Cap: ${marketCap.toLocaleString()}</p>
      <p className="coinVolume">Volume (24H): ${volume.toLocaleString()}</p>
      {priceChange < 0 ? (
        <div className="priceContainerDown">
          <i className="fas fa-angle-down fa-1x"></i>
          <p className="priceChange">{priceChange.toFixed(2)}%</p>
        </div>
      ) : (
        <div>
          <div className="priceContainerUp">
            <i className="fas fa-angle-up fa-1x"></i>
            <p className="priceChange">{priceChange.toFixed(2)}%</p>
          </div>
        </div>
      )}
    </div>
  );
};

export default Coin;
