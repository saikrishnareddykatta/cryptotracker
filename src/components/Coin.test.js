import React from "react";
import Coin from "./Coin";
import { render, screen } from "@testing-library/react";

describe("Test App Component", () => {
  const props = {
    id: "0",
    imageSrc:
      "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
    name: "Bitcoin",
    symbol: "btc",
    price: "18374.03",
    marketCap: "351733780640",
    volume: "90463295403",
    priceChange: 11.17162,
  };
  it("should test priceContainerUp when priceChange is greater than 0", () => {
    const wrapper = render(<Coin {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should test priceContainerDown when priceChange is less than 0", () => {
    props.priceChange = -11.17162;
    const wrapper = render(<Coin {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
