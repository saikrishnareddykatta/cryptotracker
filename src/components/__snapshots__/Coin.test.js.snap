// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Component should test priceContainerDown when priceChange is less than 0 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cryptoCoin"
      >
        <img
          alt="Bitcoin-0 symbol"
          class="coinLogo"
          src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
        />
        <div
          class="coinNameWrap"
        >
          <h1
            class="coinName"
          >
            Bitcoin
          </h1>
          <p
            class="coinSymbol"
          >
            BTC
          </p>
        </div>
        <p
          class="coinPrice"
        >
          $
          18374.03
        </p>
        <p
          class="coinMarketCap"
        >
          Market Cap: $
          351733780640
        </p>
        <p
          class="coinVolume"
        >
          Volume (24H): $
          90463295403
        </p>
        <div
          class="priceContainerDown"
        >
          <i
            class="fas fa-angle-down fa-1x"
          />
          <p
            class="priceChange"
          >
            -11.17
            %
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cryptoCoin"
    >
      <img
        alt="Bitcoin-0 symbol"
        class="coinLogo"
        src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
      />
      <div
        class="coinNameWrap"
      >
        <h1
          class="coinName"
        >
          Bitcoin
        </h1>
        <p
          class="coinSymbol"
        >
          BTC
        </p>
      </div>
      <p
        class="coinPrice"
      >
        $
        18374.03
      </p>
      <p
        class="coinMarketCap"
      >
        Market Cap: $
        351733780640
      </p>
      <p
        class="coinVolume"
      >
        Volume (24H): $
        90463295403
      </p>
      <div
        class="priceContainerDown"
      >
        <i
          class="fas fa-angle-down fa-1x"
        />
        <p
          class="priceChange"
        >
          -11.17
          %
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test App Component should test priceContainerUp when priceChange is greater than 0 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cryptoCoin"
      >
        <img
          alt="Bitcoin-0 symbol"
          class="coinLogo"
          src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
        />
        <div
          class="coinNameWrap"
        >
          <h1
            class="coinName"
          >
            Bitcoin
          </h1>
          <p
            class="coinSymbol"
          >
            BTC
          </p>
        </div>
        <p
          class="coinPrice"
        >
          $
          18374.03
        </p>
        <p
          class="coinMarketCap"
        >
          Market Cap: $
          351733780640
        </p>
        <p
          class="coinVolume"
        >
          Volume (24H): $
          90463295403
        </p>
        <div>
          <div
            class="priceContainerUp"
          >
            <i
              class="fas fa-angle-up fa-1x"
            />
            <p
              class="priceChange"
            >
              11.17
              %
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cryptoCoin"
    >
      <img
        alt="Bitcoin-0 symbol"
        class="coinLogo"
        src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
      />
      <div
        class="coinNameWrap"
      >
        <h1
          class="coinName"
        >
          Bitcoin
        </h1>
        <p
          class="coinSymbol"
        >
          BTC
        </p>
      </div>
      <p
        class="coinPrice"
      >
        $
        18374.03
      </p>
      <p
        class="coinMarketCap"
      >
        Market Cap: $
        351733780640
      </p>
      <p
        class="coinVolume"
      >
        Volume (24H): $
        90463295403
      </p>
      <div>
        <div
          class="priceContainerUp"
        >
          <i
            class="fas fa-angle-up fa-1x"
          />
          <p
            class="priceChange"
          >
            11.17
            %
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
